src/pcapi/core/offerers/api.py

```diff
@@ -91,3 +98,45 @@ def create_venue(venue_data: PostVenueBodyModel) -> Venue:
     link_valid_venue_to_irises(venue=venue)
 
     return venue
+
+
+def generate_and_save_api_key(offerer_id: int) -> str:
+    model_api_key, clear_api_key = generate_api_key(offerer_id)
+    repository.save(model_api_key)
+    return clear_api_key
+
+
+def generate_api_key(offerer_id: int) -> tuple[ApiKey, str]:
+    prefix_identifier = secrets.token_hex(8)
+    prefix = _create_prefix(settings.ENV, prefix_identifier)
+    clear_secret = secrets.token_hex(32)
+    key = ApiKey(
+        offererId=offerer_id, prefix=prefix, secret=bcrypt.hashpw(_encode_clear_secret(clear_secret), bcrypt.gensalt())
+    )
+
+    return key, f"{prefix}{API_KEY_SEPARATOR}{clear_secret}"
+
+
+def find_api_key(key: str) -> Optional[ApiKey]:
+    try:
+        env, prefix_identifier, clear_secret = key.split(API_KEY_SEPARATOR)
+        prefix = _create_prefix(env, prefix_identifier)
+    except ValueError:
+        # TODO: remove this legacy behaviour once we forbid old keys
+        return ApiKey.query.filter_by(value=key).first()
```

Comment:
```suggestion
        return ApiKey.query.filter_by(value=key).one_or_none()
```

... même s'il n'y a pas de contrainte d'unicité sur la colonne `value`, on va fortement espérer que les valeurs sont uniques ! :)